### ISSUE TITLE
Do indexed getters ever return null?

### ISSUE THREAD

**Issue by annevk** [heycam/webidl#532]
That'd be interesting to know for https://github.com/whatwg/dom/issues/284. At least whenever I use indexed getters for some legacy object I'm pretty sure that the properties always correspond to an object and not null. It's only the `item()` method that can return null.

(If it turns out that they cannot return null, we should allow the types of iterables and indexed getters to slightly differ. Or perhaps say that the type of the indexed getter is the same as the method declaration it might share, minus nullability.)

**Comment by bzbarsky** [comment]
Is this a question about practical usage in APIs right now?

**Comment by annevk** [comment]
Yeah, I basically want to know if we want/can make this an invariant or whether I should change the IDL in https://dom.spec.whatwg.org/#interface-nodelist to say Node? rather than Node even though it can never be null.

**Comment by bzbarsky** [comment]
OK.  So looking at Firefox, the indexed getters we have that return nullable things are:

     ChromeNodeList.item
     CSSRuleList.item
     CSSValueList.item
     DOMRectList.item
     DOMStringList.item
     DOMTokenList.item
     FileList.item
     GridLines.item
     GridTracks.item
     HTMLAllCollection anonymous getter
     HTMLCollection.item
     HTMLFormControlsCollection.item
     HTMLOptionsCollection.item
     HTMLSelectElement.item
     MediaList.item
     MimeTypeArray.item
     NamedNodeMap.item
     NodeList.item
     PaintRequestList.item
     PluginArray.item
     Plugin.item
     RadioNodeList.item
     StyleSheetList.item
     TouchList.item
     TreeColumns.getColumnAt

(not all of these are defined in specs; I did check the two obvious non-spec ones in ChromeNodeList and TreeColumns and they don't return null).

Also, I have no idea why HTMLAllCollection's anonymous indexed getter is defined to maybe return null.  It can't, obviously.

**Comment by bzbarsky** [comment]
I tried going through that list, and it does seem like for all of them we never return null from the getter.

So I guess the real question is whether we ever expect to support a list of "something-or-null"...
